index.js

```diff
@@ -240,13 +240,19 @@ FSWatcher.prototype._awaitWriteFinish = function(path, threshold, callback) {
   (function awaitWriteFinish (prevStat) {
     fs.stat(fullPath, function(err, curStat) {
       if (err) {
-        delete this._pendingWrites[path];
+        // prevent `unlink` race condition
+        this._getWatchedDir(sysPath.dirname(path))
+          .remove(sysPath.basename(path));
+        if (this._pendingWrites[path]) {  
+          this._pendingWrites[path].cancelWait();
```

Comment:
Hmm... So maybe the solution is to add the path to _pendingWrites before calling fs.stat? this way, if an unlink event will arrive, it could cancel the wait successfully. I'll try it now